src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -329,8 +332,10 @@
                                   ;; anything referencing an env or resource
                                   ;; param is purged first
                                   garbage-collect!]]
-
-        (gc-task #(apply perform-db-maintenance! write-db (remove nil? db-maintenance-tasks)))
+        ;; Run database maintenance tasks seqentially to avoid
+        ;; competition. Each task must handle its own errors.
+        (gc-task #(doseq [maintain (remove nil? db-maintenance-tasks)]
+                    (maintain write-db)))
```

Comment:
@wkalt does this look OK to you, or think I should revert it?  (I thought it was a bit more straightforward.)